### DISCUSSION TITLE
My ROMS output .NC file does not contain all the variables required, can I still use Parcels?

### DISCUSSION THREAD

**Discussion by Xyx105** [Parcels-code/Parcels#1406] [Q&A]
Hi,

I am currently trying to use OceanParcels to model particle movemtn within the Irish Sea with the output from a ROMS model. I have been trying ti use the code found in #620 but I am finding it tricky to replicate as the output .NC files I am working with do not have all the required/same variables.

Below is the details of the first timestep/ first netcdf file in the collection. I have been able to create Cs_r using another program but I am still missing crucial variables used in #620:

Dimensions:
           x          = 398
           y          = 357
           y_2        = 358
           s_rho      = 40
           ocean_time = 1     (UNLIMITED)
Variables:
    lon_rho   
           Size:       398x357
           Dimensions: x,y
           Datatype:   double
           Attributes:
                       standard_name       = 'longitude'
                       long_name           = 'longitude'
                       units               = 'degrees_east'
                       _CoordinateAxisType = 'Lon'
    lat_rho   
           Size:       398x357
           Dimensions: x,y
           Datatype:   double
           Attributes:
                       standard_name       = 'latitude'
                       long_name           = 'latitude'
                       units               = 'degrees_north'
                       _CoordinateAxisType = 'Lat'
    lon_u     
           Size:       398x358
           Dimensions: x,y_2
           Datatype:   double
           Attributes:
                       standard_name       = 'longitude'
                       long_name           = 'longitude'
                       units               = 'degrees_east'
                       _CoordinateAxisType = 'Lon'
    lat_u     
           Size:       398x358
           Dimensions: x,y_2
           Datatype:   double
           Attributes:
                       standard_name       = 'latitude'
                       long_name           = 'latitude'
                       units               = 'degrees_north'
                       _CoordinateAxisType = 'Lat'
    lon_v     
           Size:       398x358
           Dimensions: x,y_2
           Datatype:   double
           Attributes:
                       standard_name       = 'longitude'
                       long_name           = 'longitude'
                       units               = 'degrees_east'
                       _CoordinateAxisType = 'Lon'
    lat_v     
           Size:       398x358
           Dimensions: x,y_2
           Datatype:   double
           Attributes:
                       standard_name       = 'latitude'
                       long_name           = 'latitude'
                       units               = 'degrees_north'
                       _CoordinateAxisType = 'Lat'
    s_rho     
           Size:       40x1
           Dimensions: s_rho
           Datatype:   double
           Attributes:
                       long_name = 'S-coordinate at RHO-points'
                       positive  = 'up'
                       axis      = 'Z'
    ocean_time
           Size:       1x1
           Dimensions: ocean_time
           Datatype:   double
           Attributes:
                       standard_name = 'time'
                       long_name     = 'time since initialization'
                       units         = 'seconds since 1968-5-23 00:00:00'
                       calendar      = 'standard'
                       axis          = 'T'
    h         
           Size:       398x357
           Dimensions: x,y
           Datatype:   double
           Attributes:
                       long_name   = 'bathymetry at RHO-points'
                       units       = 'meter'
                       coordinates = 'lat_rho lon_rho'
                       grid        = 'grid'
                       location    = 'face'
                       field       = 'bath, scalar'
    angle     
           Size:       398x357
           Dimensions: x,y
           Datatype:   double
           Attributes:
                       long_name   = 'angle between XI-axis and EAST'
                       units       = 'radians'
                       coordinates = 'lat_rho lon_rho'
                       grid        = 'grid'
                       location    = 'face'
                       field       = 'angle, scalar'
    mask_rho  
           Size:       398x357
           Dimensions: x,y
           Datatype:   double
           Attributes:
                       long_name     = 'mask on RHO-points'
                       coordinates   = 'lat_rho lon_rho'
                       flag_values   = [0  1]
                       flag_meanings = 'land water'
                       grid          = 'grid'
                       location      = 'face'
    zeta      
           Size:       398x357x1
           Dimensions: x,y,ocean_time
           Datatype:   single
           Attributes:
                       long_name     = 'free-surface'
                       units         = 'meter'
                       coordinates   = 'lat_rho lon_rho'
                       _FillValue    = 9.999999933815813e+36
                       missing_value = 9.999999933815813e+36
                       time          = 'ocean_time'
                       grid          = 'grid'
                       location      = 'face'
                       field         = 'free-surface, scalar, series'
    mask_u    
           Size:       398x358
           Dimensions: x,y_2
           Datatype:   double
           Attributes:
                       long_name     = 'mask on U-points'
                       coordinates   = 'lat_u lon_u'
                       flag_values   = [0  1]
                       flag_meanings = 'land water'
                       grid          = 'grid'
                       location      = 'edge1'
    ubar      
           Size:       398x358x1
           Dimensions: x,y_2,ocean_time
           Datatype:   single
           Attributes:
                       long_name     = 'vertically integrated u-momentum component'
                       units         = 'meter second-1'
                       coordinates   = 'lat_u lon_u'
                       _FillValue    = 9.999999933815813e+36
                       missing_value = 9.999999933815813e+36
                       time          = 'ocean_time'
                       grid          = 'grid'
                       location      = 'edge1'
                       field         = 'ubar-velocity, scalar, series'
    u         
           Size:       398x358x40x1
           Dimensions: x,y_2,s_rho,ocean_time
           Datatype:   single
           Attributes:
                       long_name     = 'u-momentum component'
                       units         = 'meter second-1'
                       coordinates   = 'lat_u lon_u'
                       _FillValue    = 9.999999933815813e+36
                       missing_value = 9.999999933815813e+36
                       time          = 'ocean_time'
                       grid          = 'grid'
                       location      = 'edge1'
                       field         = 'u-velocity, scalar, series'
    mask_v    
           Size:       398x358
           Dimensions: x,y_2
           Datatype:   double
           Attributes:
                       long_name     = 'mask on V-points'
                       coordinates   = 'lat_v lon_v'
                       flag_values   = [0  1]
                       flag_meanings = 'land water'
                       grid          = 'grid'
                       location      = 'edge2'
    vbar      
           Size:       398x358x1
           Dimensions: x,y_2,ocean_time
           Datatype:   single
           Attributes:
                       long_name     = 'vertically integrated v-momentum component'
                       units         = 'meter second-1'
                       coordinates   = 'lat_v lon_v'
                       _FillValue    = 9.999999933815813e+36
                       missing_value = 9.999999933815813e+36
                       time          = 'ocean_time'
                       grid          = 'grid'
                       location      = 'edge2'
                       field         = 'vbar-velocity, scalar, series'
    v         
           Size:       398x358x40x1
           Dimensions: x,y_2,s_rho,ocean_time
           Datatype:   single
           Attributes:
                       long_name     = 'v-momentum component'
                       units         = 'meter second-1'
                       coordinates   = 'lat_v lon_v'
                       _FillValue    = 9.999999933815813e+36
                       missing_value = 9.999999933815813e+36
                       time          = 'ocean_time'
                       grid          = 'grid'
                       location      = 'edge2'
                       field         = 'v-velocity, scalar, series'

How would I go about using OceanParcles with a library of netcdf files structured this way?

**Comment by erikvansebille** [comment]
Thanks for reporting this, @Xyx105. I don't use ROMS myself, so can't help you. But perhaps someone else in this discussion forum could?

**Comment by JamiePringle** [comment]
I would strongly suggest posting this to the ROMS forum at https://www.myroms.org/forum/index.php . If you are using one of the ROMS forks, you may have to did a little harder.  

But as you point you, you can re-create Cs_r from the grid stretching parameters. 

Jamie
